hoomd/pytest/test_gsd.py

```diff
@@ -0,0 +1,378 @@
+import hoomd
+import gsd.hoomd
+import numpy as np
+import pytest
+from hoomd.pytest.test_snapshot import assert_equivalent_snapshots
+
+
+@pytest.fixture(scope='function')
+def hoomd_snapshot(lattice_snapshot_factory):
+    snap = lattice_snapshot_factory(particle_types=['p1', 'p2'],
+                                    n=10, a=2.0)
+
+    typeid_list = [0] * int(snap.particles.N / 2)
+    typeid_list.extend([1] * int(snap.particles.N / 2))
+    snap.particles.typeid[:] = typeid_list[:]
+    snap.particles.velocity[:] = np.tile(np.linspace(1, 2, 3),
+                                         (snap.particles.N, 1))
+    snap.particles.acceleration[:] = np.tile(np.linspace(1, 2, 3),
+                                             (snap.particles.N, 1))
+    snap.particles.mass[:] = snap.particles.N * [1]
+    snap.particles.charge[:] = snap.particles.N * [2]
+    snap.particles.diameter[:] = snap.particles.N * [0.5]
+
+    # bonds
+    snap.bonds.types = ['b1', 'b2']
+    snap.bonds.N = 2
+    snap.bonds.typeid[:] = [0, 1]
+    snap.bonds.group[0] = [0, 1]
+    snap.bonds.group[1] = [2, 3]
+
+    # angles
+    snap.angles.types = ['a1', 'a2']
+    snap.angles.N = 2
+    snap.angles.typeid[:] = [1, 0]
+    snap.angles.group[0] = [0, 1, 2]
+    snap.angles.group[1] = [2, 3, 0]
+
+    # dihedrals
+    snap.dihedrals.types = ['d1']
+    snap.dihedrals.N = 1
+    snap.dihedrals.typeid[:] = [0]
+    snap.dihedrals.group[0] = [0, 1, 2, 3]
+
+    # impropers
+    snap.impropers.types = ['i1']
+    snap.impropers.N = 1
+    snap.impropers.typeid[:] = [0]
+    snap.impropers.group[0] = [3, 2, 1, 0]
+
+    # constraints
+    snap.constraints.N = 1
+    snap.constraints.group[0] = [0, 1]
+    snap.constraints.value[0] = 2.5
+
+    # special pairs
+    snap.pairs.types = ['p1', 'p2']
+    snap.pairs.N = 2
+    snap.pairs.typeid[:] = [0, 1]
+    snap.pairs.group[0] = [0, 1]
+    snap.pairs.group[1] = [2, 3]
+
+    return snap
+
+
+def test_write(simulation_factory, device,
+               hoomd_snapshot, tmp_path):
+
+    filename = tmp_path / "temporary_test_file.gsd"
+
+    sim = simulation_factory(hoomd_snapshot)
+    hoomd.write.GSD.write(state=sim.state, mode='wb',
+                          filename=str(filename))
+
+    with gsd.hoomd.open(name=filename, mode='wb') as traj:
+        assert_equivalent_snapshots(traj, hoomd_snapshot)
+
+
+def test_write_gsd_trigger(simulation_factory, device,
```

Comment:
This is really a test on `hoomd.write.GSD` working in running simulations, not just the trigger.

Reply:
Yeah I agree, but I'm not sure I completely understood what you mean. Do you mean that we also need to add a test that uses a different trigger?

Reply:
A test that the trigger works properly doesn't need to check that all snapshot frames are equivalent. So, you don't need to spend a lot of testing code on generating different configurations. 

The trigger controls which timesteps are written, so you could have a static simulation with only a few particles, no forces, no integrator. Specify a non-trivial trigger to GSD, say Periodic(period=10, phase=5), then read in the GSD file and validate that each frame's step matches: e.g. 
```
assert [frame.configuration.step for frame in traj] = [5, 15, 25]
```